### PR TITLE
[IMP] hr_contract: make contract and work permit expiration notice periods configurable

### PR DESCRIPTION
**Pull request by ravb-odoo**
Currently the warning messages to warn for contracts and work permits that are about to expire are hardcoded to be triggered 7 and 60 days in advance respectively. With this commit, these notice periods are made to be configurable on a per-company basis.

task-3340931

### CODE REVIEW

File: addons/hr_contract/tests/test_contract.py [hunk 1]
```diff
@@ -116,3 +117,60 @@ def test_current_contract_stage_change(self):
...
+        contract1 = self.create_contract('open', 'normal', date(2023, 1, 1), date(2023, 1, 20), employee1.id)
+
+        with freeze_time('2023-01-01'):
+            self.env['hr.contract'].update_state()
+
+            mail_activity = self.env['mail.activity'].search([('res_id', '=', contract.id), ('res_model', '=', 'hr.contract')])
+            self.assertTrue(mail_activity.exists(), "There should be reminder activity as employee contract going to end soon")
+
+            mail_activity2 = self.env['mail.activity'].search([('res_id', '=', contract1.id), ('res_model', '=', 'hr.contract')])
+            self.assertFalse(mail_activity2.exists(), "There should be reminder activity as employee contract going to end soon")
```
Comment (robcoekaerts) [inline addons/hr_contract/tests/test_contract.py:171]: In this case, Shouldn't the error text read something like "There should be no reminder as the contract is not yet about to expire."?

**Comment by robcoekaerts** [comment]
Alternative suggestion for the commit message:
[IMP] hr_contract: make contract and work permit expiration notices configurable

Currently the warning messages to warn for contracts and work permits that are about to expire are hardcoded to be triggered 7 and 60 days in advance respectively. With this commit, these notice periods are made to be configurable on a per-company basis.

task-3340931

File: addons/hr_contract/models/res_config_settings.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = 'res.config.settings'
+
+    contract_expiration_notice_period = fields.Integer(string="Contract Days", related='company_id.contract_expiration_notice_period', readonly=False)
+    work_permit_expiration_notice_period = fields.Integer(string="Work Permit", related='company_id.work_permit_expiration_notice_period', readonly=False)
```
Comment (robcoekaerts) [inline addons/hr_contract/models/res_config_settings.py:11]: Can you also change the string arguments accordingly?

**Comment by robcoekaerts** [comment]
LGTM

**Review by sofiagvaladze — CHANGES_REQUESTED** [review]

**Comment by sofiagvaladze** [comment]
@ravb-odoo thanks for your work :raised_hands: 

Some minor comments :)

File: addons/hr_contract/tests/test_contract.py [hunk 3]
```diff
@@ -116,3 +117,72 @@ def test_current_contract_stage_change(self):
 
         draft_contract.state = 'draft'
         self.assertEqual(self.employee.contract_id, contract)
+
+    def test_check_multi_company_contract_expiration(self):
```
[the remark this answers is not in the document]
Comment (sofiagvaladze) [reply addons/hr_contract/tests/test_contract.py:146]: Oh I think we have a a misunderstanding :sweat_smile: I didn't mean to rewrite existing tests. We can just leave them the way they are. I was talking about the test that you added. 

But seeing the changes, maybe it was more readable prior. Just only the last changes. Sorry for the confusion :innocent:

**Comment by sofiagvaladze** [comment]
robodoo r+

**Comment by sofiagvaladze** [comment]
robodoo r+

File: addons/hr_contract/tests/test_auto_status.py [hunk 4]
```diff
@@ -45,14 +45,14 @@ def test_contract_enddate(self):
         self.assertEqual(self.contract.state, 'close')
 
     def test_contract_pending_visa_expire(self):
```
Comment (Bertrand2) [inline addons/hr_contract/tests/test_auto_status.py:47]: If we're testing the work permit, the title should reflect that as well :)
```suggestion
    def test_contract_pending_work_permit_expire(self):
```

**Pull request closed without merging**